pkgs/applications/system/coolercontrol/coolercontrol-gui.nix

```
@@ -0,0 +1,66 @@
+{ python3
+, qtbase
+, qtsvg
+, qtimageformats
+, qtwayland
+, wrapQtAppsHook
+}:
+
+{ version
+, src
+, meta
+}:
+
+python3.pkgs.buildPythonApplication {
+  pname = "coolercontrol";
+  inherit version src;
+  sourceRoot = "source/coolercontrol-gui";
+  format = "pyproject";
+
+  pythonRelaxDeps = true;
+
+  pythonRemoveDeps = [ "pyside6" ]; # resolution issue with the wheel and relaxed dependencies
+
+  nativeBuildInputs = [
+    wrapQtAppsHook
+  ] ++ (with python3.pkgs; [
+    poetry-core
+    setuptools
+    pythonRelaxDepsHook
+  ]);
+
+  buildInputs = [
+    qtbase
+    qtimageformats
+    qtsvg
+    qtwayland
+  ];
+
+  propagatedBuildInputs = with python3.pkgs; [
+    pyside6
+    apscheduler
+    matplotlib
+    numpy
+    setproctitle
+    jeepney
+    requests
+    dataclass-wizard
+  ];
+
+  postInstall = ''
+    install -Dm644 "${src}/packaging/metadata/org.coolercontrol.CoolerControl.desktop" -t "$out/share/applications/"
+    install -Dm644 "${src}/packaging/metadata/org.coolercontrol.CoolerControl.metainfo.xml" -t "$out/share/metainfo/"
+    install -Dm644 "${src}/packaging/metadata/org.coolercontrol.CoolerControl.png" -t "$out/share/icons/hicolor/256x256/apps/"
+    install -Dm644 "${src}/packaging/metadata/org.coolercontrol.CoolerControl.svg" -t "$out/share/icons/hicolor/scalable/apps/"
+  '';
+
+  dontWrapQtApps = true;
+
+  preFixup = ''
+    makeWrapperArgs+=("''${qtWrapperArgs[@]}")
+  '';
+
+  meta = meta // {
+    description = "${meta.description} (GUI)";
+  };
```

Aaaaand we should explicitly set `meta.mainProgram` for these, new expectation by `lib.getExe`:
```
nix-repl> lib.getExe coolercontrol.coolercontrold
trace: warning: getExe: Package "coolercontrold-0.17.0" does not have the meta.mainProgram attribute. We'll assume that the main program has the same name for now, but this behavior is deprecated, because it leads to surprising errors when the assumption does not hold. If the package has a main program, please set `meta.mainProgram` in its definition to make this warning go away. Otherwise, if the package does not have a main program, or if you don't control its definition, specify the full path to the program, such as "${lib.getBin foo}/bin/bar".
"/nix/store/yklmyax42ckzflkpx0w288vp2pz6wpia-coolercontrold-0.17.0/bin/coolercontrold"
```

```suggestion
    mainProgram = "coolercontrol";
  };
```